src/pages/main/components/MapModal.tsx

```diff
@@ -1,106 +1,266 @@
+                  this._$element.find('.arrow')[0].offsetHeight,
+              ],
+            ])
+          );
+        },
+
+        _isElement: function (element) {
+          return element && element[0] && element.find('.arrow')[0];
+        },
+      }
+    );
+
+    const MyBalloonContentLayout = ymaps.templateLayoutFactory.createClass(
+      `<div class="balloon" style="background-color: white; width: 300px; border-radius: 15px">
+        <div class="img-container">
+          <img src="$[properties.thumbnailUrl]"   style="width: 300px; height: 250px; object-fit: cover; border-radius: 15px 15px 0 0" alt="" />
+        </div>
+        <div class="info-container"  style="padding: 10px">
+          <div class="title" style="display: flex; align-self: center; justify-content: space-between;">
+            <div style=" font-weight: 700;margin-right: 40px; margin: 10px 30px 10px 0">
+            $[properties.balloonAdress]
+            </div>
+            <div style="display: flex; align-items: center">
+              <svg
+                xmlns="http://www.w3.org/2000/svg"
+                viewBox="0 0 32 32"
+                aria-hidden="true"
+                role="presentation"
+                focusable="false"
+                style="
+                  display: block;
+                  height: 12px;
+                  width: 12px;
+                  fill: currentcolor;
+                  margin-right: 7px;
+                "
+              >
+                <path
+                  fill-rule="evenodd"
+                  d="m15.1 1.58-4.13 8.88-9.86 1.27a1 1 0 0 0-.54 1.74l7.3 6.57-1.97 9.85a1 1 0 0 0 1.48 1.06l8.62-5 8.63 5a1 1 0 0 0 1.48-1.06l-1.97-9.85 7.3-6.57a1 1 0 0 0-.55-1.73l-9.86-1.28-4.12-8.88a1 1 0 0 0-1.82 0z"
+                ></path>
+              </svg>
+              4,73
+            </div>
+          </div>
+          <div class="accomm__title" style="margin-bottom: 4px">$[properties.balloonAccommTitle]</div>
+          <div class="accomm__price"><span>$[properties.iconCaption]</span> night</div>
+        </div>
+      </div>`
+    );
+
+    const {
+      minPrice,
+      maxPrice,
+      minRooms,
+      minPeople,
+      orderByPrice,
+      orderByPeople,
+      orderByRoom,
+      location,
+      checkInDate,
+      checkOutDate,
+    } = searchParamsAsObject;
+    const newParams = {
+      page: 1,
+      limit: maxAccommInMap || AMOUNT_PER_PAGE,
+      maxPrice: Number(maxPrice) || null,
+      minPrice: Number(minPrice) || null,
+      minRooms: Number(minRooms) || null,
+      minPeople: Number(minPeople) || null,
+      orderByPrice: orderByPrice == 'any' ? null : orderByPrice,
+      orderByPeople: orderByPeople == 'any' ? null : orderByPeople,
+      orderByRoom: orderByRoom == 'any' ? null : orderByRoom,
+      location: location || null,
+      checkInDate: checkInDate || null,
+      checkOutDate: checkOutDate || null,
+    };
+
+    const queryParamsString = Object.entries(newParams)

```

Comment:
create util for working with params. use URL constructor from js core
let url = new URL('https://google.com/search');
url.searchParams.set('q', 'test me!');

Reply:
done)